README.md
# Changelog for Weapons balance 29.12.2020

- Added ACE barrel and ACE ammo stats
- Renamed magazines to accurately fit their content

- Weapon changes:
    - PI:
        - Mk.I No.4 variants:
            - Changed name to "Lee-Enfield No.4 Mk.I"
        - Mk.I No.4 Carbine:
            - Changed name to "Lee-Enfield No.4 Mk.I Carbine"
        - Model 97 Carbine:
            - Changed name to "Model 97 (Short)"
        - RK 95s:
            - Changed names to "RK 95 TP"
        - SPR220s:
            - Changed name to "SPR 220"
        - SPR220 (Sawed Off):
            - Changed names to "SPR 220 (Sawn-off)"
        - TRG 42/F & TRG 42/F MMRS:
            - Added CBA Magwell for .338 mags

- MagazineWells:
    - CBA:
        - .300WM AICS:
            - Added 1x PI .300WM 10Rnd mag
    - Vanilla:
        - .338 MAR10:
            - Added 2x PI .338WM 5Rnd mags

- Weight balancing:
    - PI:
        - 1x .300WM 10Rnd mag from 0.36kg to 0.24kg
        - 1x .303 Lee Enfield 10Rnd clip from 0.20kg to 0.26kg
        - Both .338LM 5Rnd mags from 0.45kg to 0.32kg

# Changelog for Weapons balance 20.10.2020

- Weapon changes:
    - PI:
        - Falkor petra:
            - Added CBA Magwell for .300WM
            - Added long optic rail
            - Changed weight from 9.07kg to 4.67kg
        - Mk.I No.4:
            - Added CBA Magwell for .303 Lee Enfield
            - Changed weight from 6.35kg to 4.19kg
        - Mk.I No.4 Carbine:
            - Added optic rail
            - Changed weight from 3.63kg to 3.18kg
        - Model 97:
            - Added CBA Magwell for 1-6Rnds 12 Gauge shells
            - Changed weight from 6.35kg to 3.63kg
        - Model 97 Carbine:
            - Added CBA Magwell for 1-5Rnds 12 Gauge shells
            - Changed weight from 4.08kg to 2.95kg
        - RK 95:
            - Added CBA Magwell for 7.62x39mm mags/drum
            - Changed weight from 4.54kg to 3.70kg
        - RK 95 (Rail):
            - Added optic rail
            - Changed weight from 4.54kg to 3.88kg
        - SPR220:
            - Added CBA Magwell for 1-2Rnds 12 Gauge shells
            - Changed weight from 8.16kg to 3.63kg
        - SPR220 (Sawed Off):
            - Changed weight from 3.18kg to 2.72kg
        - TRG 42/F:
            - Added optic rail
            - Changed weight from 15.88kg to 5.22kg
        - TRG 42/F MMRS:
            - Added long optic rail
            - Changed weight from 15.88kg to 5.22kg

- MagazineWells:
    - CBA:
        - 12 Gauge, 2 Rnds:
            - Added 2x PI 2Rnd
        - 12 Gauge, 5 Rnds:
            - Added 2x PI 5Rnd
        - 12 Gauge, 6 Rnds:
            - Added 2x PI 6Rnd
        - .300WM AI:
            - Added 1x PI .300WM 10Rnd mag
        - .303 Lee Enfield:
            - Added 1x PI .303 10Rnd clip
        - .338LM:
            - Added 2x PI .338LM 5Rnd mags
        - 7.62x39 AK:
            - Added 1x PI 30Rnd mag

- Weight balancing:
    - PI:
        - 1x .300WM 10Rnd mag from 0.36kg to 0.24kg
        - 1x .303 Lee Enfield 10Rnd clip from 0.36kg to 0.20kg
        - Both .338LM 5Rnd mags from 0.45kg to 0.32kg
        - Both 12 Gauge 2Rnd shotgun shells from 0.18kg to 0.10kg
        - Both 12 Gauge 5Rnd shotgun shells from 0.23kg to 0.25kg
        - Both 12 Gauge 6Rnd shotgun shells from 0.27kg to 0.30kg
        - 1x 7.62x39mm AK 30Rnd mag from 0.54kg to 0.61kg


CHANGELOG.md
# Changelog for Weapons balance 15.5.2022

1.1.0.3
- Fixed minor bug.

# Changelog for Weapons balance 22.2.2022

1.1.0.2
- Minor cleanup.

# Changelog for Weapons balance 1.11.2021

1.1.0.1
- Some background changes, nothing much content wise.

# Changelog for Weapons balance 9.8.2021

1.1.0.0
- Added hemtt support
- Added weapon attachments compatibility

# Changelog for Weapons balance 29.12.2020

- Added ACE barrel and ACE ammo stats
- Renamed magazines to accurately fit their content

- Weapon changes:
    - PI:
        - Mk.I No.4 variants:
            - Changed name to "Lee-Enfield No.4 Mk.I"
        - Mk.I No.4 Carbine:
            - Changed name to "Lee-Enfield No.4 Mk.I Carbine"
        - Model 97 Carbine:
            - Changed name to "Model 97 (Short)"
        - RK 95s:
            - Changed names to "RK 95 TP"
        - SPR220s:
            - Changed name to "SPR 220"
        - SPR220 (Sawed Off):
            - Changed names to "SPR 220 (Sawn-off)"
        - TRG 42/F & TRG 42/F MMRS:
            - Added CBA Magwell for .338 mags

- MagazineWells:
    - CBA:
        - .300WM AICS:
            - Added 1x PI .300WM 10Rnd mag
    - Vanilla:
        - .338 MAR10:
            - Added 2x PI .338WM 5Rnd mags

- Weight balancing:
    - PI:
        - 1x .300WM 10Rnd mag from 0.36kg to 0.24kg
        - 1x .303 Lee Enfield 10Rnd clip from 0.20kg to 0.26kg
        - Both .338LM 5Rnd mags from 0.45kg to 0.32kg

# Changelog for Weapons balance 20.10.2020

- Weapon changes:
    - PI:
        - Falkor petra:
            - Added CBA Magwell for .300WM
            - Added long optic rail
            - Changed weight from 9.07kg to 4.67kg
        - Mk.I No.4:
            - Added CBA Magwell for .303 Lee Enfield
            - Changed weight from 6.35kg to 4.19kg
        - Mk.I No.4 Carbine:
            - Added optic rail
            - Changed weight from 3.63kg to 3.18kg
        - Model 97:
            - Added CBA Magwell for 1-6Rnds 12 Gauge shells
            - Changed weight from 6.35kg to 3.63kg
        - Model 97 Carbine:
            - Added CBA Magwell for 1-5Rnds 12 Gauge shells
            - Changed weight from 4.08kg to 2.95kg
        - RK 95:
            - Added CBA Magwell for 7.62x39mm mags/drum
            - Changed weight from 4.54kg to 3.70kg
        - RK 95 (Rail):
            - Added optic rail
            - Changed weight from 4.54kg to 3.88kg
        - SPR220:
            - Added CBA Magwell for 1-2Rnds 12 Gauge shells
            - Changed weight from 8.16kg to 3.63kg
        - SPR220 (Sawed Off):
            - Changed weight from 3.18kg to 2.72kg
        - TRG 42/F:
            - Added optic rail
            - Changed weight from 15.88kg to 5.22kg
        - TRG 42/F MMRS:
            - Added long optic rail
            - Changed weight from 15.88kg to 5.22kg

- MagazineWells:
    - CBA:
        - 12 Gauge, 2 Rnds:
            - Added 2x PI 2Rnd
        - 12 Gauge, 5 Rnds:
            - Added 2x PI 5Rnd
        - 12 Gauge, 6 Rnds:
            - Added 2x PI 6Rnd
        - .300WM AI:
            - Added 1x PI .300WM 10Rnd mag
        - .303 Lee Enfield:
            - Added 1x PI .303 10Rnd clip
        - .338LM:
            - Added 2x PI .338LM 5Rnd mags
        - 7.62x39 AK:
            - Added 1x PI 30Rnd mag

- Weight balancing:
    - PI:
        - 1x .300WM 10Rnd mag from 0.36kg to 0.24kg
        - 1x .303 Lee Enfield 10Rnd clip from 0.36kg to 0.20kg
        - Both .338LM 5Rnd mags from 0.45kg to 0.32kg
        - Both 12 Gauge 2Rnd shotgun shells from 0.18kg to 0.10kg
        - Both 12 Gauge 5Rnd shotgun shells from 0.23kg to 0.25kg
        - Both 12 Gauge 6Rnd shotgun shells from 0.27kg to 0.30kg
        - 1x 7.62x39mm AK 30Rnd mag from 0.54kg to 0.61kg


addons/main/CfgWeapons.hpp
class asdg_OpticRail1913_long;

class CfgWeapons {
    class Rifle;
    class Rifle_Base_F: Rifle {
        class WeaponSlotsInfo {};
    };
    class Rifle_Long_Base_F: Rifle_Base_F {
        class WeaponSlotsInfo: WeaponSlotsInfo {};
    };
    class bnae_falkor_base: Rifle_Long_Base_F { // https://falkordefense.com/petra/
        ACE_barrelLength = 558.8;
        ACE_barrelTwist = 247.142;
        magazineWell[] += {"CBA_300WM_AI","CBA_300WM_AICS"};
        class WeaponSlotsInfo: WeaponSlotsInfo {
            mass = 103;
            delete CowsSlot;
            delete MuzzleSlot;
            delete UnderBarrelSlot;
            class CowsSlot: asdg_OpticRail1913_long {};
            class MuzzleSlot: asdg_MuzzleSlot_338 {};
            class UnderBarrelSlot: asdg_UnderSlot {};
        };
    };

    class bnae_mk1_base: Rifle_Long_Base_F {
        ACE_barrelLength = 640;
        ACE_barrelTwist = 254;
        magazineWell[] += {"CBA_303B_LeeEn"};
        class WeaponSlotsInfo: WeaponSlotsInfo {
            mass = 92.4;
        };
    };
    class bnae_mk1_virtual: bnae_mk1_base {
        displayName = "Lee-Enfield No.4 Mk.I";
    };
    class bnae_mk1_t_virtual: bnae_mk1_base {
        displayName = "Lee-Enfield No.4 Mk.I (T)";
    };
    class bnae_mk1_t_camo1_virtual: bnae_mk1_t_virtual {
        displayName = "Lee-Enfield No.4 Mk.I (T) (Paint)";
    };
    class bnae_mk1_short_virtual: bnae_mk1_base {
        ACE_barrelLength = 320;
        displayName = "Lee-Enfield No.4 Mk.I Carbine";
        class WeaponSlotsInfo: WeaponSlotsInfo {
            mass = 70;
            delete CowsSlot;
            class CowsSlot: asdg_OpticRail1913 {};
        };
    };

    class bnae_m97_base: Rifle_Long_Base_F {
        magazineWell[] += {"CBA_12g_1rnd","CBA_12g_2rnds","CBA_12g_3rnds","CBA_12g_4rnds","CBA_12g_5rnds","CBA_12g_6rnds"};
        class WeaponSlotsInfo: WeaponSlotsInfo {
            mass = 80;
        };
    };
    class bnae_m97_s_virtual: bnae_m97_base {
        displayName = "Model 97 (Short)";
        magazineWell[] = {"CBA_12g_1rnd","CBA_12g_2rnds","CBA_12g_3rnds","CBA_12g_4rnds","CBA_12g_5rnds"};
        class WeaponSlotsInfo: WeaponSlotsInfo {
            mass = 65;
        };
    };

    class bnae_rk95_base: Rifle_Base_F {
        ACE_barrelLength = 420;
        ACE_barrelTwist = 240;
        magazineWell[] += {"CBA_762x39_AK","CBA_762x39_RPK"};
        class WeaponSlotsInfo: WeaponSlotsInfo {
            mass = 81.6;
            delete MuzzleSlot;
            class MuzzleSlot: asdg_MuzzleSlot_762 {};
        };
    };
    class bnae_rk95_virtual: bnae_rk95_base {
        displayName = "RK 95 TP";
        class WeaponSlotsInfo: WeaponSlotsInfo {};
    };
    class bnae_rk95_camo1_virtual: bnae_rk95_base {
        displayName = "RK 95 TP (Desert)";
    };
    class bnae_rk95r_virtual: bnae_rk95_virtual {
        displayName = "RK 95 TP (Rail)";
        class WeaponSlotsInfo: WeaponSlotsInfo {
            mass = 85.6;
            delete CowsSlot;
            class CowsSlot: asdg_OpticRail1913 {};
        };
    };
    class bnae_rk95r_camo1_virtual: bnae_rk95r_virtual {
        displayName = "RK 95 TP (Desert/Rail)";
    };

    class bnae_spr220_base: Rifle_Base_F {
        magazineWell[] += {"CBA_12g_1rnd","CBA_12g_2rnds"};
        class WeaponSlotsInfo: WeaponSlotsInfo {
            mass = 80;
        };
    };
    class bnae_spr220_virtual: bnae_spr220_base {
        displayName = "SPR 220";
    };
    class bnae_spr220_camo1_virtual: bnae_spr220_base {
        displayName = "SPR 220 (Digital)";
    };
    class bnae_spr220_so_base: bnae_spr220_base {
        class WeaponSlotsInfo: WeaponSlotsInfo {
            mass = 60;
        };
    };
    class bnae_spr220_so_virtual: bnae_spr220_so_base {
        displayName = "SPR 220 (Sawn-off)";
    };
    class bnae_spr220_so_camo1_virtual: bnae_spr220_so_base {
        displayName = "SPR 220 (Digital/Sawn-off)";
    };

    // https://en.wikipedia.org/wiki/Sako_TRG
    class bnae_trg42_base: Rifle_Long_Base_F {
        ACE_barrelLength = 690;
        ACE_barrelTwist = 254;
        magazineWell[] += {"CBA_338LM_AI","MAR10_338"};
        class WeaponSlotsInfo: WeaponSlotsInfo {
            mass = 115;
            delete CowsSlot;
            delete MuzzleSlot;
            class CowsSlot: asdg_OpticRail1913 {};
            class MuzzleSlot: asdg_MuzzleSlot_338 {};
        };
    };
    class bnae_trg42_mmrs_virtual: bnae_trg42_base {
        class WeaponSlotsInfo: WeaponSlotsInfo {
            delete UnderBarrelSlot;
            class CowsSlot: asdg_OpticRail1913_long {};
            class UnderBarrelSlot: asdg_UnderSlot {};
        };
    };
    class bnae_trg42_f_mmrs_virtual: bnae_trg42_base {
        class WeaponSlotsInfo: WeaponSlotsInfo {
            delete UnderBarrelSlot;
            class CowsSlot: asdg_OpticRail1913_long {};
            class UnderBarrelSlot: asdg_UnderSlot {};
        };
    };
};
